wavelength-plugins/splatoon.js

```diff
@@ -0,0 +1,248 @@
+			splatProfile.weapon = target;
+			Db.splatoon.set(user.userid, splatProfile);
+			return this.sendReply(`Your Splatoon 2 weapon has been set to "${target}".`);
+		},
+
+		randweapon: "randomweapon",
+		randomweapon: function (target, room, user) {
+			if (!this.runBroadcast()) return;
+			return this.sendReplyBox(`<strong>Randomly Generated Weapon:</strong> ${weapons[Math.floor(Math.random() * weapons.length)]}`);
+		},
+
+		splatfest: {
+			enable: "on",
+			start: "on",
+			new: "on",
+			make: "on",
+			on: function (target, room, user) {
+				if (!this.can("ban", null, room)) return this.errorReply(`You must be a Room Moderator or higher to use this command.`);
+				if (room.id !== "splatoon") return this.errorReply(`This command only works in the Splatoon room.`);
+				let targets = target.split(",");
+				for (let u in targets) targets[u] = targets[u].trim();
+				let team1 = targets[0];
+				let team2 = targets[1];
+				if (!team1 || !team2) return this.parse("/splatoonhelp");
+				if (SPLATFEST.active) return this.errorReply(`Splatfest is already active.`);
+				SPLATFEST.active = true;
+				if (Rooms("splatoon")) {
+					Rooms("splatoon").addRaw(`${WL.nameColor(user.name, true)} has enabled Splatfest. The teams of this Splatfest are: ${team1} and ${team2}.`);
+				}
+				// Set Splatfest Teams
+				SPLATFEST.alpha = toId(team1);
+				SPLATFEST.bravo = toId(team2);
+			},
+
+			disable: "off",
+			end: "off",
+			cancel: "off",
+			remove: "off",
+			off: function (target, room, user) {
+				if (!this.can("ban", null, room)) return this.errorReply(`You must be a Room Moderator or higher to use this command.`);
+				if (room.id !== "splatoon") return this.errorReply(`This command only works in the Splatoon room.`);
+				if (!SPLATFEST.active) return this.errorReply(`Splatfest is not currently active.`);
+				SPLATFEST.active = false;
+				let splatfestUsers = Db.splatoon.keys();
+				for (let u in splatfestUsers) {
```

Comment:
```js
for (let u of splatfestusers) {
   let splatProfile = Db.splatoon.get(u, {ranks: {}});
   delete splatProfile.splatfest;
   Db.splatoon.set(u, splatProfile);
}
```

Reply:
gonna merge that idea w/ @HoeenCoder's to use an if statement before the delete